### ISSUE TITLE
searching for slurp isn't great

### ISSUE THREAD

**Issue by jdv** [perl6/doc#2491]
## The problem

Searching for "slurp":

1.  Turns up results for "sleep".
2.  Doesn't include /language/io which is more like what I was looking for.

**Comment by JJ** [comment]
I guess you mean typing slurp in the search slot. I see it returns slurp-rest as the first result, and somewhat down the line, slurp as a routine. If you click on that slurp, it goes to [this page](https://docs.perl6.org/routine/slurp). Is that what you mean?

**Comment by jdv** [comment]
No.  I mean when I search for slurp I want to see a synopsis in there like /language/io does.  The /routine/slurp page is virtually useless  to me.

**Comment by JJ** [comment]
Those pages are reference, and are generated automatically from the page that describes the class. We can hardly call them *useless*, and there's certainly a group of people that has worked to make that description as useful as possible. I would appreciate if you would refrain from using that language. If it's less than awesome, it might help pointing to the tutorial page, but tutorial and reference are two completely different things and the usual way is indexing methods and routines pointing to the reference, not the tutorial, page.

**Comment by treyharris** [comment]
@jdv I think there may be an argument that it makes sense here (and perhaps in `lines`, `read`, `readchar` and maybe even `open`, though that's a stretch) to defy the precedent that @JJ has correctly pointed out that we don't, in general, want "reference" pages pointing to "tutorial" pages, only the other way around. That argument is: there aren't a lot of imperative languages (even among the dynamic VHLL's) that prefer these higher-level, more functional-ish idioms for input, and we want to steer folks who may naturally, as programmers of other languages, assume a way to do things that's sub-optimal but will (painfully) work. Having new programmers miss delightful features of Perl 6 by attempting to write another language _while referring to the docs to learn how to do so_ is a doc fail, IMO. (That's why we fairly frequently have "if you want to do X, you probably should look at Y instead" commentary in the reference.)

The routine docs are auto-generated in strict lexicographic order, so one may (somewhat safely) assume that the entry for `slurp` from [doc/Type/IO.pod6](/perl6/doc/blob/0b63543c58056e2d6086bebfc517d9c2402f7ea1/doc/Type/IO.pod6#L300-L325) as visible at https://docs.perl6.org/type/IO#sub_slurp will remain in the top slot. (I _thought_ there was a way to insert some top-of-routine-page blurb somewhere, but I must be thinking of another doc repo I work with...)

So if you think there's an _especial_ need to point at the [IO guide page](/perl6/doc/blob/master/doc/Language/io-guide.pod6) or a particular section via a link to make the `slurp` content more fully grokkable, I suggest you add the pointer in the way you think would have helped you (I'd imagine no more than a sentence would be required?) to the lines linked above.

(That said, we definitely don't need _each_ of those routine reference sections to repeat one another with links to a non-canonical tutorial; one from IO should be sufficient.)
